cmd/keymasterd/certgen.go

```diff
@@ -146,6 +146,39 @@ func (state *RuntimeState) certGenHandler(w http.ResponseWriter, r *http.Request
 	}
 }
 
+// returns 3 values, if the key is valid, if the key is not valid, the text reason why and and error if it was an internal error
+
+func getValidSSHPublicKey(userPubKey string) (ssh.PublicKey, string, error) {
```

Comment:
that's in interesting return convention ... i don't fully understand why the string and error are both being returned (and not just always an error?)